core/bin/zksync_server/src/main.rs

```diff
@@ -177,7 +180,13 @@ fn main() -> anyhow::Result<()> {
         return Ok(());
     }
 
-    node.build(opt.components.0)?.run(observability_guard)?;
+    let built_node = node.build(opt.components.0)?;
```

Comment:
The task description says "after ensuring that **all** components can be built". If this is what we want, it would make sense to ignore `opt.components` and instead provide the full list of components (maybe, make the command-line arg [conflict with](https://docs.rs/clap/latest/clap/struct.Arg.html#method.conflicts_with) `components`.

Reply:
I don't think it's what we want, I incorrectly wrote "all", each env can have slightly different components, my CI script will first gather all components existing on given env and run this validator only for those components, imo it's much less error-prone